### PR TITLE
[PyPI publishing] feat: automate the process of pypi publication to some extent.

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

This PR adds utilities to semi-automate the process of publishing a new release on PyPI and notifying our internal Slack channel about it. 

## Synopsis

[`setup.py`](https://github.com/huggingface/diffusers/blob/main/setup.py) clearly lays out the steps needed in order for us to prepare a new library and publish it on PyPI. Steps from 1 - 6 need careful inspection and might often require manual intervention. But steps 7 - 9 and making an announcement on Slack about the release don't require it IMO and can be systematically automated. 

## Design

1. As mentioned earlier, the PR assumes that we need some level of babysitting till step 6 of the publishing process. So, the workflow, `pypi_publish.yaml` starts after it. More specifically, it gets triggered after a tag is created and pushed. It then does the following:
  * Checks out the latest release branch.
  * Builds the wheel and distribution. 
  * Pushes to the Test PyPI server. 
  * Runs a small import test (as prescribed in `setup.py`). 
  * If the above test succeeds, pushes to the main PyPI. 

2. The step that comes after PyPI publishing is publishing the release on our GitHub with the release notes. We then usually (should) notify in an internal Slack channel about the release. This is handled by `notify_slack_about_release.yml`. This workflow gets triggered after a release is published on our GitHub. 

## Considerations

* Both the workflows mentioned above have "workflow_dispatch" included as their additional triggers so that they can be run manually when needed. 
* I have extensively tested this design with a dummy repository: https://github.com/sayakpaul/blossom/. You can find the actions [here](https://github.com/sayakpaul/blossom/actions). The released package on PyPI is [here](https://pypi.org/project/blossoms-sayak/). Dummy Slack channel for testing: `#temp-notification-bot-test` where. 

@LysandreJik @yiyixuxu @DN6 WDYT?

### CODE REVIEW

**Review by LysandreJik — COMMENTED** [review]
If others maintainers like this workflow, SGTM!

File: .github/workflows/pypi_publish.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      
+      - name: Publish to the test PyPI
+        env:
+          TWINE_USERNAME: ${{ secrets.TEST_PYPI_USERNAME }}
+          TWINE_PASSWORD: ${{ secrets.TEST_PYPI_PASSWORD }}
+        run: twine upload dist/* -r pypitest --repository-url=https://test.pypi.org/legacy/    
+
+      - name: Test installing diffusers and importing
+        run: |
+          pip install blossoms_sayak && pip uninstall blossoms_sayak -y
```
Comment (yiyixuxu) [inline .github/workflows/pypi_publish.yaml:68]: what is blossoms_sayak?

File: .github/workflows/notify_slack_about_release.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: Notify Slack about a release
+
+on:
+  workflow_dispatch:
+  release:
```
Comment (yiyixuxu) [inline .github/workflows/notify_slack_about_release.yml:5]: 👍👍👍

**Review by yiyixuxu — COMMENTED** [review]
thanks!

File: .github/workflows/pypi_publish.yaml (hunk 1 above)
Comment (sayakpaul) [reply .github/workflows/pypi_publish.yaml:68]: Sorry, testing residue. Will fix.

**Review by yiyixuxu — APPROVED** [review]
thanks!

**Comment by sayakpaul** [comment]
@DN6 LMK what you think. Would like to test-drive this with the upcoming release 🚗

**Review by DN6 — APPROVED** [review]

**Pull request merged**